### DISCUSSION TITLE
Access righs to class/struct members

### DISCUSSION THREAD

**Discussion by freedxmgxd** [carbon-language/carbon-lang#1706] [Language design]
Since the choice to make the members of a class public by default, I still think about a new keyword access right and design to Class. Since are often used getters to make private members readable, I starting to think about a "read" keyword to make some members only readable by external access.
We can have something like :
`class NewClass{
private var private_member: String;
read var only_readable_member: String;
var public_member: String;
}`

In this way dispense the necessity to declare a whole new method getter to make a parameter only readable by external access.

**Comment by MichaelBonnet** [comment]
This seems a little over-engineering to me. While I definitely see the use in such a keyword, writing getters is fairly trivial.

**Reply by freedxmgxd** [reply]
I dunno how much cost was to implement this, even in the compiler time, then the engineering necessary to this, but access rights similar to file-system permission seems good to me.

**Comment by geoffromer** [comment]
The "properties" feature in [Swift](https://docs.swift.org/swift-book/LanguageGuide/Properties.html) and other languages might be a good point of comparison here.
